packages/data/src/graphql/fetch.ts

```diff
@@ -12,19 +12,36 @@ export type GqlFetchResult<T> = {
 export const fetchGraphQl = async <T, V = Record<string, unknown>>({
   query,
   variables,
+  network,
 }: {
   query: DocumentNode | string;
   variables?: V;
+  network?: Network | '';
 }): Promise<GqlFetchResult<T>> => {
-  try {
-    const client = new GraphQLClient(GRAPHQL_ENDPOINT);
-    return {
-      data: await client.request<T>(query, variables),
-    };
-
-  } catch (error:unknown) {
-    return {
-      error: getErrorMessage(error),
-    };
+
+  const endpointReady =  mbjs?.keys?.isSet || network?.length > 0; 
+
+  let graphqlEndpoint = mbjs?.keys.graphqlUrl ?? '';
+
+  if (network && network.length > 0) {
```

Comment:
What's the motivation for empty string here vs null or undef?

Reply:
fixed